mitiq/rem/measurement_result.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Defines MeasurementResult, a result obtained by measuring qubits on a
+quantum computer."""
+from dataclasses import dataclass
+from typing import cast, Iterable, List
+
+import numpy as np
+
+
+Bitstring = List[int]
```

Comment:
I'm not sure how to document a type, but yes you're understanding is right. It does not mean we measure every qubit at the end. An subset of qubits can be measured.

I did add a docstring with an example to `MeasurementResult`.

Reply:
Apologies for jumping in, but Sphinx by default looks for comments starting with `#:` to document constants. From the perspective of Sphinx, a type hint is just another Python constant, so that should work.

Reply:
Thanks @cgranade! I'll add this in follow-up PR, and good to know in general :)